.github/workflows/kanister-image-build.yaml

```diff
@@ -0,0 +1,39 @@
+name: build-kanister-image
+
+on:
+  pull_request:
+    branches:
+    - master
```

Comment:
Is this workflow going to run on every PR?

Reply:
We should only run this when the Dockerfile has changed on a `push` request to `master` and not on a pull request.
Check out https://github.com/tj-actions/changed-files

Reply:
made the workflow as `on-demand` only.